presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -178,11 +179,10 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
             return new FixedSplitSource(ImmutableList.of());
         }
 
-        // get buckets from first partition (arbitrary)
-        Optional<HiveBucketFilter> bucketFilter = layout.getBucketFilter();
+        Optional<HiveBucketFilter> bucketFilter = isBucketExecutionEnabled(session) ? layout.getBucketFilter() : Optional.empty();
 
         // validate bucket bucketed execution
-        Optional<HiveBucketHandle> bucketHandle = layout.getBucketHandle();
+        Optional<HiveBucketHandle> bucketHandle = isBucketExecutionEnabled(session) ? layout.getBucketHandle() : Optional.empty();
```

Comment:
I think we should set `layout$bucketHandle` to `empty()` if bucket execution is disabled much earlier. Say, in https://github.com/prestodb/presto/blob/cca4e4a628e3d7159ac029b8388acb55b44428dc/presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java#L123.

After we have this check in the HivePartitionManager (that is called from the HiveMetadata#getTableLayouts: https://github.com/prestodb/presto/blob/e629ad1879316b010df3c31db6ec259dc55c35b9/presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java#L1653), we can remove the old check from the `HiveMetadata#getTableLayout`: https://github.com/prestodb/presto/blob/e629ad1879316b010df3c31db6ec259dc55c35b9/presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java#L1689.